src/mlpack/methods/fastmks/fastmks_impl.hpp

```diff
@@ -420,7 +505,7 @@ void FastMKS<KernelType, MatType, TreeType>::Search(
   // Dual-tree implementation.
   Timer::Stop("computing_products");
 
-  Search(referenceTree, k, indices, kernels);
+  Search(std::move(referenceTree), k, indices, kernels);
```

Comment:
There's no need for a move here, this will destroy the model's tree after the call.  Passing `referenceTree` as a `const Tree&` is just fine.